Crash on reading file names in utf8

ERROR >>>
$ edwood 
panic: internal error: fileinsert

goroutine 1 [running]:
main.(*File).Insert(0xc0001cc100, 0x160, 0xc0000a19ac, 0x1, 0x1)
	/home/dmytro/go/src/github.com/rjkroege/edwood/file.go:109 +0x3d9
main.(*Text).Columnate(0xc0001ca128, 0xc0001f2000, 0x22d, 0x350, 0xc000191300, 0x22d, 0x22d)
	/home/dmytro/go/src/github.com/rjkroege/edwood/text.go:257 +0x4b6
main.(*Text).Load(0xc0001ca128, 0x0, 0xc0000160a4, 0x16, 0x1, 0x0, 0x0, 0x0)
	/home/dmytro/go/src/github.com/rjkroege/edwood/text.go:322 +0x9d0
main.readfile(0xc0000f6780, 0xc0000160a4, 0x16)
	/home/dmytro/go/src/github.com/rjkroege/edwood/acme.go:195 +0xc2
main.main()
	/home/dmytro/go/src/github.com/rjkroege/edwood/acme.go:157 +0xb0c

Fix:
In text.go (Lines 236-238):
			dl := names[j]
			t.file.Insert(q1, []rune(dl))
			q1 += (len(dl))
len of string and []rune could be different, so: 
			dl := []rune(names[j])
			t.file.Insert(q1, dl)
			...

--- a/dat.go
+++ b/dat.go
@@ -201,12 +201,6 @@ type Xfid struct {
 
 type RangeSet []Range
 
-type Dirlist struct {
-	r   []rune
-	nr  int
-	wid int
-}
-
 type Expand struct {
 	q0    int
 	q1    int

--- a/text.go
+++ b/text.go
@@ -233,9 +233,9 @@ func (t *Text) Columnate(names []string, widths []int) {
 	q1 = 0
 	for i := 0; i < nrow; i++ {
 		for j := i; j < len(names); j += nrow {
-			dl := names[j]
-			t.file.Insert(q1, []rune(dl))
-			q1 += (len(dl))
+			dl := bytetorune([]byte(names[j]))
+			t.file.Insert(q1, dl)
+			q1 += len(dl)
 			if j+nrow >= len(names) {
 				break
 			}

--- a/util.go
+++ b/util.go
@@ -62,6 +62,11 @@ func restoremouse(w *Window) bool {
 	return false
 }
 
+func bytetorune(s []byte) []rune {
+	r, _, _ := cvttorunes(s, len(s))
+	return r
+}
+
 // TODO(flux) The "correct" answer here is return unicode.IsNumber(c) || unicode.IsLetter(c)
 func isalnum(c rune) bool {
 	/*

